### ISSUE TITLE
Saved PNGs are upside down

### ISSUE THREAD

**Issue by vorg** [deanm/plask#36]
Using  `this.canvas.writeImage('png', 'test.png');` produces image upside down. The following code will draw red square in the top left corner but it's in the bottom left in the resulting png.

Using build 5d26b850f55b43ca54f0de8aebb5a28cdefc2e92

```javascript
var plask = require('plask');

plask.simpleWindow({
  type: '2d',
  init: function() {
    this.paint.setFill();
    this.paint.setColor(255, 0, 0, 255);
    this.canvas.drawRect(this.paint, 0, 0, 50, 50);
    this.canvas.writeImage('png', 'test.png');
  }
})

```

**Comment by deanm** [comment]
Strange... Is it the same for OpenGL contexts?

**Comment by vorg** [comment]
OpenGL is ok

**Comment by deanm** [comment]
: (

![screen shot 2015-02-05 at 18 54 33](https://cloud.githubusercontent.com/assets/56582/6065797/763d11f8-ad68-11e4-8eaf-d3ce62f0cd1d.png)

**Comment by deanm** [comment]
I will look at writing out PNGs from Skia without FreeImage, will clean up the code a lot since now it also has to handle unpremultiplication, etc.

**Comment by deanm** [comment]
Fixed in 7193e1e5de92dba4af8f05c468a4b3e9b9370db7 ?

**Comment by vorg** [comment]
Fixed.

Curious at which point did it break though. It was fine in Plask v1 before.

Cheers

On Thu, Feb 5, 2015 at 9:45 PM, Dean McNamee <notifications@github.com>
wrote:

> Fixed in 7193e1e
> <https://github.com/deanm/plask/commit/7193e1e5de92dba4af8f05c468a4b3e9b9370db7>
> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/deanm/plask/issues/36#issuecomment-73135421>.
>



-- 
Marcin Ignac
http://marcinignac.com
@marcinignac

**Comment by deanm** [comment]
I broke it in e882f0925f09adc546dda915719dd50ccb0ecbea
